### ISSUE TITLE
[FEATURE] Add support of admonitions

### ISSUE THREAD

**Issue by tobiasdiez** [Zettlr/Zettlr#532]
## Description
Admonitions are a simple way to have shout-out boxes. They were originally introduced by `mkdocs` (I think) and are usually written in markdown as
```
!!! note
    You should note that the title will be automatically capitalized.
```

It would be nice if Zettlr would support them by properly rendering such notes.

![render example](https://raw.githubusercontent.com/qjebbs/vscode-markdown-extended/master/images/admonition-demo1.png)

## Use cases
- Highlight important information
- It would be fantastic to obtain a global list of all admonition (by type). This would make it possible to use it to collected related things that are specific to one page (e.g. one does a review of some published paper and got an idea for a further project from this; add an adminition "idea" to the page that contains notes about this paper and the overview page can give a nice summary of all ideas one got from various papers).

## Additional Information
Description:
https://squidfunk.github.io/mkdocs-material/extensions/admonition/
https://github.com/qjebbs/vscode-markdown-extended#admonition
https://python-markdown.github.io/extensions/admonition/

Implementation:
https://github.com/vsch/flexmark-java/wiki/Admonition-Extension

Related:
https://asciidoctor.org/docs/asciidoc-writers-guide/#admonitions

**Comment by nathanlesage** [comment]
Several questions:

1. Is this compatible to Markdown Editors that do not feature something like that? I'm talking especially concerning non-monospace spacing
2. Does Pandoc support that?
3. Could this be related to this issue here: #192? If so, I'd like to combine these issues to save time and not become mental in the process :)

**Comment by tobiasdiez** [comment]
1. For editors not supporting admonitions, they are just normal text so I don't see any problem.
2. Pandoc does not yet support them https://github.com/jgm/pandoc/issues/2610
3. While not the same, there is definitely some overlap. For example, I could imagine that todo's are implemented using
```markdown
!!! todo 
    - [ ] Task 1
    - [ ] Task 2
```

**Comment by nathanlesage** [comment]
1. Yes, that for sure. I just wanted to make sure they don't interfere with other highlighting! :)
2. Looks as if one can enable that, but it seems a difficult process
3. Mh, looks intereesting!

Let's keep this and see how it's going!

**Comment by tobiasdiez** [comment]
@stale this issue is still not resolved.

**Comment by nathanlesage** [comment]
Well, the question is how useful this would be for other people? There hasn't been any activity besides this, and I'm pretty hesitant in adding this to the list of stuff I need to do myself … unless you'd like to tackle this?

**Comment by tobiasdiez** [comment]
I agree it's not the most urgent thing to have, but some form of blocks should be supported in my opinion (I still need to check what's the most used form of blocks for pandoc). You can assign this issue to me if you want.

General remark: I know that having such a large number of open issues can be a burden and feel overwhelming. Moreover, it seems Zettlr got quite some attention recently resulting in even more bug and feature reports. Thus, it is natural that you want to decrease the number to keep an overview. Naturally, stale bot is a nice help for this. However, there is one big caveat in my opinion: it works quite well for bugs that have been resolved but forgotten. But it's not as effective in my opinion for feature requests. Just because a suggestion is old, it doesn't mean that it is no longer wanted or wouldn't reach a large user base. In my opinion, there is no real problem in having a huge backlog of things that need to be done at some point. The real issue is that you as a maintainer loose sight of what's important / urgent. From experience this means quite often that you only fix/implement what people reported recently, instead of focusing on some maybe more important issues that were reported some time ago (that's definitely not the case for the admonition support, was meant more as a general remark).

If I may suggest some other approach: categorize issues in groups of "high/low priority" and "help wanted". Issues in the first two categories are things that you want to work on yourself or at least are concerned with important things. Issues labeled with "help wanted" are then things that you don't want to implement yourself but would accept PRs if somebody else feels like spending some time on it. In this way, you keep track of what's important for the project, but still keep feature requests that might inspire other contributions as well. (This is of course just a suggestion, Zettlr is your baby!)

**Comment by nathanlesage** [comment]
I fully agree. The stale bot was just my very first attempt at doing *something* to begin tackling all the issues. For instance, it helped me identify some things and re-categorize them which I've already forgotten. But also it took a huge mental responsibility from me, because now I have what psychoanalysis calls the *big Other* (the bot) to refer back to if something is about to be closed, giving me more sovereignty in deciding which issues I'd like to keep and which ones to discard (e.g. on this issue, I was now able to say "Well, as of now there's only one person wanting this, so maybe it's not that important…?")

But, on the other hand, I'd like to encourage PRs from other people, which only recently have begun coming in, and about which I am very happy!

I think, the most ideal solution that would put my mind fully at ease would be to have people that I could add as members to this repository so that I'm not the only one with rights to close issues, merge PRs, attach labels, and manage all the stuff in here, but then I feel that I can't demand of other people to have this responsibility and therefore I'm too afraid to ask people if they want to become members of the repo :D

**Comment by nathanlesage** [comment]
@tobiasdiez Would you mind triaging this either onto the 1.7.x project or the 1.8.x one? Depending on scope and how long you'd like to take with this one!

**Comment by tobiasdiez** [comment]
Apparently I don't have write access to the projects, so here it is: 1.8.x

**Comment by nathanlesage** [comment]
> Apparently I don't have write access to the projects

Now you … might? GitHub is somewhat unspecific concerning the rights, but I just increased everything by a level! (+ did the project assignment, but feel free to test it out!)

**Comment by tobiasdiez** [comment]
Now it works, thanks!

**Comment by Kangie** [comment]
+1 for this feature.

I'd definitely like to see a 'standard' way to put them into markdown - the mkdocs syntax looks pretty good. To be properly useful though there needs to be some way to handle this on export to HTML/PDF/DOCX. I don't know how easy that would be -  [jgm/pandoc#2610](https://github.com/jgm/pandoc/issues/2610) seems to indicate that Pandoc doesn't support this yet.

Not having an easy way to denote Info/Warning/Error admonitions in the doco that I write is a big pain point at the moment!

**Comment by nathanlesage** [comment]
> Not having an easy way to denote Info/Warning/Error admonitions in the doco that I write is a big pain point at the moment!

The Zettlr documentation misuses blockquotes for that currently, but yes, it's somewhat a pain … what is the MkDocs syntax? Because … well, if there is syntax, I'd like to switch Zettlr docs to that as well :D

**Comment by Kangie** [comment]
I'm also abusing blockquotes. :'(

[Here](https://squidfunk.github.io/mkdocs-material/reference/admonitions/) is a pretty good reference as to the syntax that mkdocs uses - it's the same syntax that @tobiasdiez has indicated above.

The TL;DR version though is that an admonition should begin with `!!!` and any text (apart from the title, which is contained in quotes on the same line) should be indented four spaces, e.g.:

```
!!! note
    This is a note admonition.

!!! warning
    This is a warning admonition
    It has several lines.

!!! error "Error Title"
    This is an error admonition. It also has a title, that will replace the text "Error".
    It will retain the error icon and colour though.
```

**Comment by nathanlesage** [comment]
No, have a look at the roadmap: https://github.com/orgs/Zettlr/projects/2?query=is%3Aopen+sort%3Aupdated-desc

I won't be able to implement this anytime soon; but I'm happy to review a PR.

**Comment by connect-mahdi** [comment]
Hi! The Zettlr is **amazing**

How can we help to further the implementation of this feature?

Could it be achieved with a custom CSS?

Markor Android App did implement it that way I think...

**Comment by nathanlesage** [comment]
> Could it be achieved with a custom CSS?

Unfortunately, that won't help. What we need is custom parsers for the various formats of admonitions and allow users to switch between them.

Essentially, there are two major flavors:

```
!!! note
    This is an MKdocs style admonition
```

and:

```
> [!NOTE]
> This is a GitHub style admonition
```

Both would need to be supported by Zettlr. Then, if that's implemented, we would need custom Lua filters that also produce valid admonitions during export.

So it's unfortunately not quite straight forward.
